external_data/task_functions.py

```diff
@@ -0,0 +1,534 @@
+import filecmp
+import os
+import zipfile
+from ast import literal_eval
+
+import pandas as pd
+import requests
+from airflow.models import Variable
+from elasticsearch import helpers
+from elasticsearch_dsl import connections
+
+ELASTIC_PASSWORD = Variable.get("ELASTIC_PASSWORD")
+ELASTIC_URL = Variable.get("ELASTIC_URL")
+ELASTIC_USER = Variable.get("ELASTIC_USER")
+ENV = Variable.get("ENV")
+
+
+def preprocess_colter_data(
+    data_dir,
+    **kwargs,
+) -> None:
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+    # Process Régions
+    df = pd.read_csv(
+        "https://www.data.gouv.fr/fr/datasets/r/619ee62e-8f9e-4c62-b166-abc6f2b86201",
+        dtype=str,
+        sep=";",
+    )
+    df = df[df["exer"] == df.exer.max()][["reg_code", "siren"]]
+    df = df.drop_duplicates(keep="first")
+    df = df.rename(columns={"reg_code": "colter_code_insee"})
+    df["colter_code"] = df["colter_code_insee"]
+    df["colter_niveau"] = "region"
+
+    # Cas particulier Corse
+    df.loc[df["colter_code_insee"] == "94", "colter_niveau"] = "particulier"
+    dfcolter = df
+
+    # Process Départements
+    df = pd.read_csv(
+        "https://www.data.gouv.fr/fr/datasets/r/2f4f901d-e3ce-4760-b122-56a311340fc4",
+        dtype=str,
+        sep=";",
+    )
+    df = df[df["exer"] == df["exer"].max()]
+    df = df[["dep_code", "siren"]]
+    df = df.drop_duplicates(keep="first")
+    df = df.rename(columns={"dep_code": "colter_code_insee"})
+    df["colter_code"] = df["colter_code_insee"] + "D"
+    df["colter_niveau"] = "departement"
+
+    # Cas Métropole de Lyon
+    df.loc[df["colter_code_insee"] == "691", "colter_code"] = "69M"
+    df.loc[df["colter_code_insee"] == "691", "colter_niveau"] = "particulier"
+    df.loc[df["colter_code_insee"] == "691", "colter_code_insee"] = None
+
+    # Cas Conseil départemental du Rhone
+    df.loc[df["colter_code_insee"] == "69", "colter_niveau"] = "particulier"
+    df.loc[df["colter_code_insee"] == "69", "colter_code_insee"] = None
+
+    # Cas Collectivité Européenne d"Alsace
+    df.loc[df["colter_code_insee"] == "67A", "colter_code"] = "6AE"
+    df.loc[df["colter_code_insee"] == "67A", "colter_niveau"] = "particulier"
+    df.loc[df["colter_code_insee"] == "67A", "colter_code_insee"] = None
+
+    # Remove Paris
+    df = df[df["colter_code_insee"] != "75"]
+
+    dfcolter = pd.concat([dfcolter, df])
+
+    # Process EPCI
+    df = pd.read_excel(
+        "https://www.collectivites-locales.gouv.fr/files/2022/epcisanscom2022.xlsx",
+        dtype=str,
+        engine="openpyxl",
+    )
+    df["colter_code_insee"] = None
+    df["siren"] = df["siren_epci"]
+    df["colter_code"] = df["siren"]
+    df["colter_niveau"] = "epci"
+    df = df[["colter_code_insee", "siren", "colter_code", "colter_niveau"]]
+    dfcolter = pd.concat([dfcolter, df])
+
+    # Process Communes
+    URL = "https://www.data.gouv.fr/fr/datasets/r/42b16d68-958e-4518-8551-93e095fe8fda"
```

Comment:
`URL` variable is too generic in my option. Especially because you use so many urls here.